tracer/test/Datadog.Trace.Security.Unit.Tests/IAST/HashBasedDeduplicationTests.cs

```diff
@@ -0,0 +1,151 @@
+// <copyright file="HashBasedDeduplicationTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.Iast;
+using Xunit;
+
+namespace Datadog.Trace.Security.Unit.Tests.Iast
+{
+    public class HashBasedDeduplicationTests
+    {
+        [Fact]
+        public void GivenTheSameVulnerabilityInstance_WhenAddedToDeduplication_OnlyOneIsStored()
```

Comment:
Shouldn't you disable parallelism on those tests at they rely on the `HashBasedDeduplication` static cache? Couldn't we swicth to a singleton with an internal .ctor to ease testing?

Reply:
Yes, it would be a good idea to prevent these tests from race conditions. I will change it.

Reply:
This is not a correct usage of `CollectionDefinition` 🙂  But you can just delete it anyway, as this is the only type that uses it - tests within a single class are _always_ run in serial. You only need to disable parallelization when you have _multiple test classes_ using the same shared objects. Which you avoid wherever possible 😄 

> Couldn't we switch to a singleton with an internal .ctor to ease testing

+1 on this 🙂

Reply:
Yes, I added it just in case that in the future some tests could interfere, since this class uses the static class. Inside the class, there is no problem... Anyway, I can just remove it and add it if in the future there are other tests that could interfere which, as you said, would not be desirable anyway.

Reply:
>  tests within a single class are always run in serial

:til: ok sorry for the noise

Reply:
I have implemented the singleton pattern for HashBasedDeduplication. Thanks for the suggestion!

Reply:
Seeing as there's no _setter_ needed on `HashBasedDeduplication`, could we reduce the singleton implementation to just this? 🤔 

```csharp
internal class HashBasedDeduplication
{
    public static HashBasedDeduplication Instance { get; } = new();
}
```

Reply:
Yes, that is a good one. That would be much more simple and I cannot think about any disadvantage over the previous solution. I will update the code.  Thanks!